### ISSUE TITLE
Check trait argument names

### ISSUE THREAD

**Issue by sanpii** [rust-lang/rust-clippy#17592]
### What it does

Check if the argument names are the same in trait definition and in implementation.

### Advantage

- avoid confusing with bad argument naming;
- consistant argument name across different part of code.

### Drawbacks

- Maybe I don’t want to name my argument as the same as in the trait definition ;
- Just check if an argument as the same name of another.

### Example

```rust
#![allow(unused)]

trait Foo {
    fn foo(user: &str, password: &str);
}

struct Bar;

impl Foo for Bar {
    fn foo(password: &str, user: &str) {
    }
}
```

Could be written as:

```rust
#![allow(unused)]

trait Foo {
    fn foo(user: &str, password: &str);
}

struct Bar;

impl Foo for Bar {
    fn foo(user: &str, password: &str) {
    }
}
```

### Comparison with existing lints

I don’t find lint for this.

**Comment by hashcatHitman** [comment]
I believe this lint already exists. Please see [`clippy::renamed_function_params`] and clarify if you were looking for something different. [Example on Rust Playground] (top right corner, Tools > Clippy).

[`clippy::renamed_function_params`]: https://rust-lang.github.io/rust-clippy/master/index.html#renamed_function_params
[Example on Rust Playground]: https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=82ed11f3ed947dc7b574aa54d6c29c1b
